src/ucc/api/ucc.h

```diff
@@ -1381,19 +1381,36 @@ typedef enum ucc_memory_type {
     UCC_MEMORY_TYPE_UNKNOWN = UCC_MEMORY_TYPE_LAST
 } ucc_memory_type_t;
 
+/**
+ *  @ingroup UCC_COLLECTIVES_DT
+ */
 typedef struct ucc_coll_buffer_info_v {
-    void             *buffer;
-    ucc_count_t      *counts;
-    ucc_aint_t       *displacements;
-    ucc_datatype_t    datatype;
-    ucc_memory_type_t mem_type;
+    void             *buffer; /*!< Starting address of the send/recv buffer */
+    ucc_count_t      *counts; /*!< Array of counts of type @ref ucc_count_t
+                                describing the total number of elements */
+    ucc_aint_t       *displacements; /*!< Displacement array of team size and
```

Comment:
@manjugv do we want to specify that displacements are given in terms of "counts of datatypes" rather than in bytes? Or is it evident enough?

Reply:
We haven't used bytes in any of the descriptions, so, I think it is obvious that we are talking about datatype. If you think there is ambiguity. I can change the counts to this "Array of counts of type @ref ucc_count_t describing the total number of **@ref ucc_datatype_t** elements"